### PR TITLE
Create Requirements Checklist schema for HOTH API

### PR DESCRIPTION
**Pull request by juliusfitzhugh-ccpo**
Create `Requirements Checklist` schema.

Minor adjustment of `IncrementalFundingPlan` schema
- Fix amount `type`s to be `number`s and add `format`

Ticket: AT-8361

### CODE REVIEW

File: hoth_api.yaml [hunk 1]
```diff
@@ -456,11 +458,14 @@ components:
...
+          format: float
         initialAmount:
-          type: float
+          type: number
+          format: float
         remainingAmount:
-          type: float
+          type: number
+          format: float
         fundingIncrements:
```
Comment (jasonburkert-ccpo) [inline hoth_api.yaml:469]: I'd mention this adjustment in the PR description if intentional

File: hoth_api.yaml [hunk 2]
```diff
@@ -568,6 +573,103 @@ components:
...
+        A Task Order (Alternative) Contracting Officer Representative (COR/ACOR) information
+      type: object
+      properties:
+        name:
+          type: string
+        email:
+          type: string
+        phoneNumber:
+          type: string
+        dodaa:
```
Comment (jasonburkert-ccpo) [inline hoth_api.yaml:671]: Department of Defense Activity Address Code (DoDAAC) ?

**Review by jasonburkert-ccpo — CHANGES_REQUESTED** [review]

File: hoth_api.yaml (hunk 1 above)
Comment (juliusfitzhugh-ccpo) [reply hoth_api.yaml:469]: I started to write something in the commit, but decided not. I'll add it in PR description.

File: hoth_api.yaml (hunk 2 above)
Comment (jasonburkert-ccpo) [reply hoth_api.yaml:671]:
```suggestion
        dodaac:
```

Comment (juliusfitzhugh-ccpo) [reply hoth_api.yaml:671]: Added as `description`.

Comment (jasonburkert-ccpo) [reply hoth_api.yaml:672]: I think the property name needs a `c` at the end.  I added the code suggestion later, you may not have seen it.

**Review by jasonburkert-ccpo — APPROVED** [review]
Looks good!

**Pull request merged**